### PR TITLE
Explode to voices

### PR DESCRIPTION
**Pull request by Rockettwo**
This is my attempt to explode voices from the top selected staff to the staffs below. It extends the previous implementation, so now it checks whether there is something in another voice than voice one and if so, it does explode every voice to a new staff. If there is just voice one in the staff, it explodes every note of the chords in this voice to separate staffs below. 

The feature of explode is that it deletes the content in the source staff. This feature was also achieved with the new explode tool by making some changes to the previous cloneVoice method. But the explode for voices doesn't overwrite the previous content of the staffs like it is done by the actual. It looks for the first free voice in the staff and if there isn't a free voice the actual voice won't be exploded. In the most common use cases voices are exploded to empty staffs below the selection, so this won't be a problem at all I think. 

Tests are coming, I'll copy the new directory implemented [here](https://github.com/musescore/MuseScore/pull/2717) and add them there.

### CODE REVIEW

File: libmscore/cmd.cpp [hunk 1]
```diff
@@ -2660,62 +2660,127 @@ void Score::cmdExplode()
...
+
+      int lTick = endMeasure->endTick();
+      bool voice = false;
+
+      for (Measure* m = startMeasure; m && m->tick() != lTick; m = m->nextMeasure()) {
+            if (m->hasVoices(srcStaff)) {
+                        voice = true;
+                        break;
+                        }
+                  }
```
Comment (Jojo-Schmitz) [inline libmscore/cmd.cpp]: indetation of this block seems 6 spaces too deep

File: mtest/CMakeLists.txt [hunk 2]
```diff
@@ -173,6 +173,7 @@ subdirs (
         libmscore/element
         libmscore/exchangevoices
         libmscore/hairpin
+	libmscore/implode_explode
```
Comment (Jojo-Schmitz) [inline mtest/CMakeLists.txt]: indentation

File: mtest/libmscore/CMakeLists.txt [hunk 3]
```diff
@@ -12,8 +12,7 @@
 #=============================================================================
 
 subdirs(
-      album barline beam breath chordsymbol clef clef_courtesy compat concertpitch copypaste
-          copypastesymbollist cursor dynamic earlymusic element exchangevoices hairpin instrumentchange join keysig layout links parts measure midi      midimapping note plugins repeat selectionfilter selectionrangedelete spanners split splitstaff timesig tools transpose tuplet text
+      album barline beam breath chordsymbol clef clef_courtesy compat concertpitch copypaste copypastesymbollist cursor dynamic earlymusic element exchangevoices hairpin instrumentchange join keysig layout links parts measure midi
```
Comment (Jojo-Schmitz) [inline mtest/libmscore/CMakeLists.txt]: Where's 'midimapping note plugins repeat selectionfilter selectionrangedelete spanners split splitstaff timesig transpose tuplet text' (I understood that you moved tools)

**Comment by lasconic** [comment]
Hey @Rockettwo, two of your PR has been merged. Now implode and explode PR need a rebase. Can you take a look?

**Comment by Rockettwo** [comment]
@lasconic, thanks for noticing me. I did rebase the two but I'll have a look whether they are really working properly together with the merged PR.

**Comment by lasconic** [comment]
Thanks! As you might have seen I did some small changes in the UI and the logic. In particular, there was a bug when making parts for a piano score and unselecting all voices in the G clef. https://github.com/musescore/MuseScore/commit/67b189f8dfccc2691a4e3f3bf8e8cd445454c963

**Comment by lasconic** [comment]
Could you rebase one more time, sorry... I will merge it next.

**Comment by Jojo-Schmitz** [comment]
mtest/libmscore/implode_explode/explode1-ref.mscx needs to get adjusted to pass mtests

**Comment by Jojo-Schmitz** [comment]
Looks like a failed rebase?

**Comment by Jojo-Schmitz** [comment]
Looks good now, time to merge?

**Comment by MarcSabatella** [comment]
I had totally forgotten about this code when I started a forum discussion about explode behavior for voices - see https://musescore.org/en/node/264654

I think there is widespread agreement explode should handle multiple voices, but the consensus in that thread is that we need to handle it *in addition* to handling chords, not *instead of*.  That is, we need to handle music that mixes and matches the two approaches and does reasonable things in either case.

What was the use case for the code here?  Would an algorithm like we are settling on in that forum thread make sense for this use case?

**Pull request merged**